Assignment6.Rmd

```diff
@@ -72,12 +73,55 @@ table(M2$certified, M2$predict2)
 
 ```
 
+I feel like the 2nd tree is better, because despite the 1st tree's accuracy in prediciting those who are not certified, it made a lot of false positives, which is an issue. 
+
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
+library(dplyr)
+library(tidyr)
+library(stringr)
+library(rpart)
+
+M3 <- read.csv("student.record.csv", header = TRUE)
+names(M3) <- c("MAJOR_3", "MAJOR_2", "MAJOR_1", "HSGPA", "ACT_ENGL", "ACT_MATH", "ACT_READ", "ACT_SCIRE", "ACT_COMP", "SAT_VERB", "SAT_MATH", "SAT_COMP", "SEX", "STDNT_GROUP1", "STDNT_GROUP2", "MAJOR1_DEPT", "MAJOR2_DEPT", "MAJOR3_DEPT", "ANONID", "ADMIT_TERM", "MAJOR1_TERM", "MAJOR2_TERM", "MAJOR3_TERM")
+
+#find students with BA in DEGREE_1 title & label accordingly
+majors2 <- select(M3, 3:13, 16, 19)
+
+majors_ART <- majors2
+
+majors_ART$MAJOR_1 <- as.character(majors_ART$MAJOR_1)
+
+majors_ART <- majors_ART %>% 
+  select(MAJOR_1:SEX, ANONID) %>% 
+  filter_all(any_vars(str_detect(MAJOR_1, pattern = "BA")))
+
+majors_ART$DEG_TYPE <- "art"
+
+#find students with BS in DEGREE_1 title & label accordingly
+majors_SCIENCE <- majors2
+majors_SCIENCE$MAJOR_1 <- as.character(majors_SCIENCE$MAJOR_1)
+
+majors_SCIENCE <- majors_SCIENCE %>% 
+  select(MAJOR_1:SEX, ANONID) %>% 
+  filter_all(any_vars(str_detect(MAJOR_1, pattern = "BS")))            
+
+majors_SCIENCE$DEG_TYPE <- "science"
+
+#Bind BA and BS students together
+majors_ALL <- bind_rows(majors_ART, majors_SCIENCE)
+majors_ALL_math <- majors_ALL %>% 
+  select(ANONID, MAJOR_1, DEG_TYPE, ACT_MATH, SAT_MATH, HSGPA)
+
+#Do math scores predict BA or BS? spoiler alert, yes. 
+c.tree4 <- rpart(as.factor(DEG_TYPE) ~ ACT_MATH + SAT_MATH, method = "class", data = majors_ALL_math)
+
+printcp(c.tree4)
```

Comment:
Good job! I like the way that you process/deal with the data to prepare for later modeling. This task asks you to compare at least two different models, so you could go further to choose different predictors/features and compare measures from the result.